modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/function/SsviVolatilityFunction.java

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.volatility.smile.function;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.Property;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+
+/**
+ * Surface Stochastic Volatility Inspired (SSVI) formula.
+ * <p>
+ * Reference: Gatheral, Jim and Jacquier, Antoine. Arbitrage-free SVI volatility surfaces. arXiv:1204.0646v4, 2013. Section 4.
+ */
+@BeanDefinition
+public final class SsviVolatilityFunction
+    extends VolatilityFunctionProvider<SsviFormulaData> implements ImmutableBean, Serializable {
+
+  /**
+   * Default implementation. 
+   */
+  public static final SsviVolatilityFunction DEFAULT = new SsviVolatilityFunction();
+  
+  /** SSVI volatility description diverge for theta -> 0. Lower bound for which time to expiry is accepted. */
+  public static final double MIN_TIME_TO_EXPIRY = 1.0E-3;
+
+  @Override
+  public double getVolatility(double forward, double strike, double timeToExpiry, SsviFormulaData data) {
+    ArgChecker.isTrue(timeToExpiry > MIN_TIME_TO_EXPIRY, "time to expiry must not be zero to be able to compute volatility");
+    double volatilityAtm = data.getSigma();
+    double rho = data.getRho();
+    double eta = data.getEta();
+    double theta = volatilityAtm * volatilityAtm * timeToExpiry;
+    double phi = eta / Math.sqrt(theta);
+    double k = Math.log(strike / forward);
+    double w = 0.5 * theta * (1.0d + rho * phi * k + Math.sqrt(1.0d + 2 * rho * phi * k + phi * k * phi * k));
+    return Math.sqrt(w / timeToExpiry);
+  }
+
+  @Override
+  public ValueDerivatives getVolatilityAdjoint(double forward, double strike, double timeToExpiry, SsviFormulaData data) {
```

Comment:
Because the content of ValueDerivatives is model specific, it is useful if brief javadoc is added especially for the array "derivatives".